projects/plugins/jetpack/modules/publicize/publicize-jetpack.php

```diff
@@ -740,26 +992,41 @@ function enhaced_twitter_cards_site_tag( $tag ) {
 		$connections = $this->get_connections( 'twitter' );
 		foreach ( $connections as $connection ) {
 			$connection_meta = $this->get_connection_meta( $connection );
-			if ( 0 == $connection_meta['connection_data']['user_id'] ) {
-				// If the connection is shared
+			if ( 0 === (int) $connection_meta['connection_data']['user_id'] ) {
```

Comment:
Adding another comment from earlier. 
```
public function is_global_connection(  $connection ) {

   return empty(  $connection['connection_data']['user_id']  );
}
```
if ( !is_global_connection ( $connection_meta ) )
```
or 

simply `empty( $connection_meta['connection_data']['user_id'] ) ` would suffice as well, we can skip the type casting too since empty works on '0' as well.